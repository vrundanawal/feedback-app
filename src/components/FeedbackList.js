import { motion, AnimatePresence } from "framer-motion";
import { useContext } from "react";
import FeedbackItem from "./FeedbackItem";
import feedbackContext from "../context/FeedbackContext";

const FeedbackList = ({ handleDelete }) => {
  const { feedback } = useContext(feedbackContext);
  //console.log(feedback);
  if (!feedback || feedback.length === 0) {
    return <p data-testid="error-message">No FeedBack is available</p>;
  }

  //not to add any animation liabrary for this one. woking without add framer-motion package

  // return (
  //   <div className="feedback-list">
  //     {feedback.map((item) => (
  //       <FeedbackItem key={item.id} item={item} handleDelete={handleDelete} />
  //     ))}
  //   </div>
  // );

  return (
    <div className="feedback-list">
      <AnimatePresence>
        {feedback.map((item) => (
          <motion.div
            key={item?.id}
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            exit={{ opacity: 0 }}
          >
            <FeedbackItem
              key={item?.id}
              item={item}
              handleDelete={handleDelete}
            />
          </motion.div>
        ))}
      </AnimatePresence>
    </div>
  );
};

export default FeedbackList;
