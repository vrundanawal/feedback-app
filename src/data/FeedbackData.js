const FeedbackData = [
  {
    id: 1,
    rating: 10,
    text: "Using context Api Item 1.",
  },
  {
    id: 2,
    rating: 9,
    text: "Using context Api Item 2",
  },
  {
    id: 3,
    rating: 8,
    text: "Using context Api Item 3",
  },
  {
    id: 4,
    rating: 9,
    text: "Using context Api Item 3",
  },
];

export default FeedbackData;
