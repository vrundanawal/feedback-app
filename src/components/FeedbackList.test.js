import "@testing-library/jest-dom";

import FeedbackList from "./FeedbackList";
//import userEvent from "@testing-library/user-event";
import { shallow } from "enzyme";
import FeedbackData from "../data/FeedbackData";

describe("<FeedbackList/>", () => {
  it("should render the feedback List component", () => {
    expect(FeedbackList).toBeTruthy();
  });
});

test("should test TodoLists component with default state of array", () => {
  const wrapper = shallow(<FeedbackList feedback={FeedbackData} />);
  expect(wrapper).toMatchSnapshot();
});
