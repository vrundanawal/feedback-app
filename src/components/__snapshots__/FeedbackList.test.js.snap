// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test TodoLists component with default state of array 1`] = `
<div
  className="feedback-list"
>
  <AnimatePresence>
    <ForwardRef(MotionComponent)
      animate={
        Object {
          "opacity": 1,
        }
      }
      exit={
        Object {
          "opacity": 0,
        }
      }
      initial={
        Object {
          "opacity": 0,
        }
      }
      key="1"
    >
      <FeedbackItem
        item={
          Object {
            "id": 1,
            "rating": 10,
            "text": "Lorem ipsum dolor sit amet consectetur adipisicing elit. consequuntur vel vitae commodi alias voluptatem est voluptatum ipsa quae.",
          }
        }
        key="1"
      />
    </ForwardRef(MotionComponent)>
    <ForwardRef(MotionComponent)
      animate={
        Object {
          "opacity": 1,
        }
      }
      exit={
        Object {
          "opacity": 0,
        }
      }
      initial={
        Object {
          "opacity": 0,
        }
      }
      key="2"
    >
      <FeedbackItem
        item={
          Object {
            "id": 2,
            "rating": 9,
            "text": "Lorem ipsum dolor sit amet consectetur adipisicing elit. consequuntur vel vitae commodi alias voluptatem est voluptatum ipsa quae.",
          }
        }
        key="2"
      />
    </ForwardRef(MotionComponent)>
    <ForwardRef(MotionComponent)
      animate={
        Object {
          "opacity": 1,
        }
      }
      exit={
        Object {
          "opacity": 0,
        }
      }
      initial={
        Object {
          "opacity": 0,
        }
      }
      key="3"
    >
      <FeedbackItem
        item={
          Object {
            "id": 3,
            "rating": 8,
            "text": "Lorem ipsum dolor sit amet consectetur adipisicing elit. consequuntur vel vitae commodi alias voluptatem est voluptatum ipsa quae.",
          }
        }
        key="3"
      />
    </ForwardRef(MotionComponent)>
  </AnimatePresence>
</div>
`;
